agents/wiregrid_kikusui/kikusui_agent.py

```diff
@@ -0,0 +1,603 @@
+import sys
+import os
+import argparse
+import time
+import numpy as np
+import traceback
+
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import TimeoutLock
+
+# add PATH to ./src directory
+this_dir = os.path.dirname(__file__)
+sys.path.append(
+        os.path.join(this_dir, 'src'))
+
+# import classes
+import pmx
+import command
+from common import openlog, writelog
+
+
+class KikusuiAgent:
+    def __init__(self, agent, kikusui_ip, kikusui_port):
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.take_data = False
+        self.kikusui_ip = kikusui_ip
+        self.kikusui_port = int(kikusui_port)
+
+        self.position_path = '/data/wg-data/position.log'
+        self.action_path = '/data/wg-data/action/'
```

Comment:
I assume to mount a local directory on /data/wg-data in wiregrid_encoder and wiregrid_kikusui containers.
The position.log should be shared between the two agents because the kikusui agent needs to check the current position recorded by the encoder agent.
Maybe, there is a way to retrieve the current position recorded by the encoder agent. If we can do that, please let me know how to do that.

Reply:
Yes, the way to do this is to use the [session.data](https://ocs.readthedocs.io/en/develop/developer/session-data.html). So the kikusui agent will need to create a client for the encoder agent, check the status of the `acq()` Process, and inspect the `session.data` object for the current position. (The encoder will of course have to put the position in `session.data`.)

The kikusui Agent will need to be told what the encoder agent's instance-id is, and you'll likely want to protect against the encoder agent not being online when the kikusui agent starts the client.

Reply:
I believe this is resolved in the changes [here](https://github.com/simonsobs/socs/pull/183/files/a28bab332b63cdd72ebe6889edc02ea94f163403..2e4b112ceee85c5d6e72d459a9a4a8af138ba1ae).